ipatests/test_integration/test_acme.py

```diff
@@ -252,3 +260,39 @@ def test_disable_acme_service(self):
             ['curl', '--fail', self.acme_server],
             ok_returncode=22,
         )
+
+
+class TestACMEwithExternalCA(TestACME):
+    """Test the FreeIPA ACME service with external CA"""
+
+    num_replicas = 0
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        # cache the acme service uri
+        acme_host = f'{IPA_CA_RECORD}.{cls.master.domain.name}'
+        cls.acme_server = f'https://{acme_host}/acme/directory'
+
+        # install packages before client install in case of IPA DNS problems
+        if not skip_certbot_tests:
+            tasks.install_packages(cls.clients[0], ["certbot"])
+        if not skip_mod_md_tests:
+            tasks.install_packages(cls.clients[0], ["mod_md"])
```

Comment:
This is identical to what is in the overridden `install` method.  It should be extracted to a classmethod in the superclass, e.g.:
```python
@classmethod
def prepare_acme_clients(cls):
    # the highlighted stuff
```

Then you only need to call `cls.prepare_acme_clients()` from each version of the `install` classmethod.

Reply:
Done